protocol.go

```diff
@@ -21,7 +21,7 @@ type Listener struct {
 // may be speaking the Proxy Protocol. If it is, the RemoteAddr() will
 // return the address of the client instead of the proxy address.
 type Conn struct {
-	bufReader         *bufio.Reader
+	BufReader         *bufio.Reader
```

Comment:
Since this is exported, please add a descriptive comment with [an example of a use-case](https://github.com/pires/go-proxyproto/pull/54#issuecomment-732936977). Also, please add a test or two.